### PR TITLE
Added check for author of issue against issue approval.

### LINKED ISSUE #62: Amend approve issue feature to disallow author of issue from approving an issue
## Description
As a developer,
I need to amend the feature to approve issues,
so that author of the issue can't approve it.

### Update [Required]
- [ ] Amend approve issue feature to disallow author of issue from approving an issue.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by sammy1997**
# Description
Added a check so that the author of an issue can't approve the same issue

Fixes #62 

# Type of Change:
- Code
- Bug fix (a non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)

# How Has This Been Tested?
from_slack
authour_check

# Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: slack_functions.py [hunk 1]
```diff
@@ -194,3 +205,25 @@ def open_issue_slack(data):
         send_message_to_channels(channel_id, MESSAGE.get('success_issue', ''))
     else:
         send_message_to_channels(channel_id, MESSAGE.get('error_issue', ''))
+
+def get_detailed_profile(uid):
+    body = {'user': uid, 'include_labels': True}
+    get_profile_response = requests.post(get_user_profile_info_url, data=body, headers=headers_legacy_urlencoded)
+    profile_response_json = get_profile_response.json()
```
Comment (sara-02) [inline slack_functions.py]:
```python
profile_response_json=requests.post(get_user_profile_info_url, data=body, headers=headers_legacy_urlencoded).json()
```
will save us using one unused variable

File: slack_functions.py [hunk 2]
```diff
@@ -194,3 +205,25 @@ def open_issue_slack(data):
...
+def get_github_username_profile(profile):
+        custom_fields = profile.get('fields', {})
+        github_profile_present = False
+        github_id = ""
+        for key in custom_fields:
+            github_link = custom_fields.get(key, {}).get('value', '')
+            if 'github.com/' in github_link:
+                github_profile_present = True
+                github_id = github_link.split('github.com/')[1]
+                return {'github_profile_present': github_profile_present, 'github_id': github_id}
```
Comment (sara-02) [inline slack_functions.py]: we can do
 ```python
return {'github_profile_present': True, 'github_id': github_id}
```
and get rid of `github_profile_present ` variable

File: slack_functions.py [hunk 3]
```diff
@@ -51,48 +51,59 @@ def approve_issue_label_slack(data):
...
+            return
         else:
-            #Wrong format of command was used
-            send_message_ephimeral(channel_id, uid, MESSAGE.get('correct_approve_format',''))
+            github_id = get_github_username.get('github_id', '')
+            if result.get('is_maintainer', False):
+                params = data.get('text','')
+                if params != '' and len(params.split(' ')) == 2:
+                    issue_author = get_issue_author(org_repo_owner, params.split(' ')[0], params.split(' ')[1])
+                    print(issue_author, github_id)
```
Comment (sara-02) [inline slack_functions.py]: We may want to remove this before merging.

**Comment by sammy1997** [comment]
@sara-02 Done

**Comment by sammy1997** [comment]
@sara-02 I have pushed once more due to a small error from codacy. Please review and merge it.

**Pull request merged**